go/libraries/doltcore/sqle/writer/prolly_table_writer.go

```diff
@@ -153,25 +162,72 @@ func (w *prollyTableWriter) Close(ctx *sql.Context) error {
 
 // StatementBegin implements TableWriter.
 func (w *prollyTableWriter) StatementBegin(ctx *sql.Context) {
-	// todo(andy)
 	return
 }
 
 // DiscardChanges implements TableWriter.
 func (w *prollyTableWriter) DiscardChanges(ctx *sql.Context, errorEncountered error) error {
-	// todo(andy)
-	return nil
+	err := w.primary.Discard(ctx)
+	for _, secondary := range w.secondary {
+		sErr := secondary.Discard(ctx)
+		if sErr != nil && err == nil {
+			err = sErr
+		}
+	}
+	return err
 }
 
 // StatementComplete implements TableWriter.
 func (w *prollyTableWriter) StatementComplete(ctx *sql.Context) error {
-	// todo(andy)
-	return nil
+	err := w.primary.Commit(ctx)
+	for _, secondary := range w.secondary {
+		sErr := secondary.Commit(ctx)
+		if sErr != nil && err == nil {
+			err = sErr
+		}
+	}
+	return err
 }
 
 // WithIndexLookup implements TableWriter.
 func (w *prollyTableWriter) WithIndexLookup(lookup sql.IndexLookup) sql.Table {
-	panic("prolly table writer does not yet support index lookups")
+	idx := index.IndexFromIndexLookup(lookup)
+	return prollyFkIndexer{
+		writer: w,
+		index:  idx,
+		pRange: index.ProllyRangesFromIndexLookup(lookup)[0],
+	}
+}
+
+// Reset puts the writer into a fresh state, updating the schema and index writers according to the newly given table.
+func (w *prollyTableWriter) Reset(ctx context.Context, sess *prollyWriteSession, tbl *doltdb.Table, sch schema.Schema) error {
```

Comment:
I'm missing some detail about why this is necessary. Do we do this for DDL / adding foreign keys? 
In those cases we should probably be discarding the write session and creating a new one